aws/aws.go

```diff
@@ -211,11 +226,83 @@ func (s *ec2Ops) waitAttachmentStatus(
 		fmt.Sprintf("Invalid volume object for volume %s", volumeID), "")
 }
 
-func (s *ec2Ops) Name() string { return "aws" }
+func (s *awsOps) Name() string { return "aws" }
+
+func (s *awsOps) InstanceID() string { return s.instance }
+
+func (s *awsOps) InspectInstance(instanceID string) (*cloudops.InstanceInfo, error) {
+	inst, err := DescribeInstanceByID(s.ec2, instanceID)
+	if err != nil {
+		return nil, err
+	}
 
-func (s *ec2Ops) InstanceID() string { return s.instance }
+	instInfo := &cloudops.InstanceInfo{
+		CloudResourceInfo: cloudops.CloudResourceInfo{
+			Name:   instanceID,
```

Comment:
You can get the human readable instance Name from the tags
inst.Tags["Name"] will have the actual name of the instance, if it exists. We can resort to ID if there is no Name set.

Reply:
added